cmd/harmony/main.go

```diff
@@ -367,6 +369,59 @@ func setupNodeAndRun(hc harmonyconfig.HarmonyConfig) {
 		}
 	}
 
+	// prune rewards from archival node
+	shouldPrune := hc.PruneReward != nil && hc.PruneReward.TakeAction && // asked
+		(hc.General.IsBeaconArchival || (hc.General.IsArchival && hc.General.ShardID == 0)) // eligible
+	if shouldPrune {
+		chain := currentNode.Beaconchain()
+		curNum := chain.CurrentBlock().NumberU64()
+		if curNum == 0 {
+			fmt.Println("No rewards to prune because we are at genesis block")
+			os.Exit(1)
+		}
+		chainConfig := chain.Config()
+		// max number to be last block of epoch "AggregatedRewardEpoch - 1"
+		var maxNum uint64
```

Comment:
Actually we want to prune to the last block because even with aggregated reward logic, validatorWrapper may still change by other actions such as delegation, undelegation, collect rewards and many more. So many history records may still exist after aggregated reward epoch, though much less than before.